### PR TITLE
upgrade react-navigation

### PR DESCRIPTION
**Pull request by rasom**
Animation is a bit smoother and nav is kind of ~30ms faster on slower device. Not a big difference, but anyway better to keep it up to date.

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/routing/core.cljs [hunk 1]
```diff
@@ -116,6 +117,9 @@
    (cond->
     (merge {:headerMode        "none"
             :cardStyle         {:backgroundColor (when (or platform/ios? platform/desktop?) :white)}
+            #_:transitionConfig
```
Comment (flexsurfer) [inline src/status_im/ui/screens/routing/core.cljs:120]: leftovers?

File: src/status_im/ui/screens/navigation.cljs [hunk 2]
```diff
@@ -135,7 +135,7 @@
 (handlers/register-handler-fx
  :navigate-to-clean
  (fn [cofx [_ view-id params]]
-   (navigate-to-clean cofx view-id params)))
+   (navigate-to-cofx cofx view-id params)))
```
Comment (flexsurfer) [inline src/status_im/ui/screens/navigation.cljs:138]: no clean anymore?

**Review by flexsurfer — APPROVED** [review]

**Review by yenda — APPROVED** [review]
Iirc there is some code for safe area on iphoneX is it still relevant?

**Comment by rasom** [comment]
@yenda why it shouldn't be relevant?

File: src/status_im/ui/screens/routing/core.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/ui/screens/routing/core.cljs:120]: yep

File: src/status_im/ui/screens/navigation.cljs (hunk 2 above)
Comment (rasom) [reply src/status_im/ui/screens/navigation.cljs:138]: yes, it crashes now and seems to work fine with `navigate-to-cofx`. Will see how tests will pass

**Comment by churik** [comment]
@rasom would you mind to rebase it and I will trigger e2e again?

**Comment by rasom** [comment]
@churik done

**Comment by churik** [comment]
Tested on IOS 11.4 and on Android 8.
- general app navigation
- /send, /request in chat
- swipe and force touch gestures
- search in chats
- logout and force logout actions 
- preview privacy mode
- share a chat
- signing messages in DApp 

Tested briefly on Android 6 (Samsung Galaxy S7) and Android 7 (Samsung Galaxy S8+) on BrowserStack.
Waiting for @Serhy  to check on real IPhone X and for @asemiankevich on real Android 7.
@rasom If you have smth to add for testing - let me know

**Comment by churik** [comment]
@rasom and new conflicts appear...Can you resolve?

**Comment by rasom** [comment]
@churik done

**Comment by Serhy** [comment]
No regressions with iPhoneX.

**Comment by asemiankevich** [comment]
android 7 looks fine

**Pull request merged**